docs/source/custom_sockets.rst

```diff
@@ -0,0 +1,214 @@
+.. _custom-sockets:
+
+=========================
+Custom Sockets in Syncano
+=========================
+
+``Syncano`` provides possibility of creating the custom sockets. It means that there's a possibility
+to define a very specific endpoints in syncano application and use them as normal api calls.
+Currently custom sockets allow only one dependency - script. This mean that on the backend side
+each time the api is called - the script is executed and result from this script is returned as a result of the
+api call.
+
+Creating a custom socket
+------------------------
+
+There are two methods of creating the custom socket. First: use the helpers objects defined in Python Libray.
```

Comment:
That's still the one and only method. Library simply provides some helpers for it. Indicating that it is separate somehow and is overall a different method - can lead to some confusion.  Even I'm confused after reading this ;) especially when it says about describing raw format below and then goes on to using python helpers. I think we can just skip this paragraph and move on to an example about how to actually create it